services/motion/builtin/builtin.go

```diff
@@ -430,3 +400,85 @@ func (ms *builtIn) GetPose(
 		supplementalTransforms,
 	)
 }
+
+// PlanMoveOnMap returns the plan for MoveOnMap to execute
+func (ms *builtIn) PlanMoveOnMap(
+	ctx context.Context,
+	componentName resource.Name,
+	destination spatialmath.Pose,
+	slamName resource.Name,
+	extra map[string]interface{},
+) ([][]referenceframe.Input, base.KinematicBase, error) {
+	// get the SLAM Service from the slamName
+	localizer, ok := ms.localizers[slamName]
+	if !ok {
+		return nil, nil, resource.DependencyNotFoundError(slamName)
+	}
+
+	// assert localizer as a slam service and get map limits
+	slamSvc, ok := localizer.(slam.Service)
+	if !ok {
+		return nil, nil, fmt.Errorf("cannot assert localizer of type %T as slam service", localizer)
+	}
+
+	// gets the extents of the SLAM map
+	limits, err := slam.GetLimits(ctx, slamSvc)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// create a KinematicBase from the componentName
+	component, ok := ms.components[componentName]
+	if !ok {
+		return nil, nil, resource.DependencyNotFoundError(componentName)
+	}
+	kw, ok := component.(base.KinematicWrappable)
+	if !ok {
+		return nil, nil, fmt.Errorf("cannot move component of type %T because it is not a KinematicWrappable Base", component)
+	}
+	kb, err := kw.WrapWithKinematics(ctx, localizer, limits)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pointCloudData, err := slam.GetPointCloudMapFull(ctx, slamSvc)
+	if err != nil {
+		return nil, nil, err
+	}
+	octree, err := pointcloud.ReadPCDToBasicOctree(bytes.NewReader(pointCloudData))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if extra == nil {
+		extra = make(map[string]interface{})
+	}
+	extra["planning_alg"] = "rrtstar"
+
+	// get current position
+	inputs, err := kb.CurrentInputs(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	ms.logger.Debugf("base position: %v", inputs)
+
+	dst := referenceframe.NewPoseInFrame(referenceframe.World, spatialmath.NewPoseFromPoint(destination.Point()))
+
+	f := kb.ModelFrame()
+	fs := referenceframe.NewEmptyFrameSystem("")
+	if err := fs.AddFrame(f, fs.World()); err != nil {
+		return nil, nil, err
+	}
+
+	worldState, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{referenceframe.NewGeometriesInFrame(referenceframe.World, []spatialmath.Geometry{octree})}, nil)
```

Comment:
This line is too long.  Has the linter stopped catching things like this?

Reply:
I didn't see any lint errors for it but I'll fix